include/rocksdb/db.h

```diff
@@ -1024,20 +1024,22 @@ class DB {
   // Simpler versions of the GetApproximateSizes() method above.
   // The include_flags argumenbt must of type DB::SizeApproximationFlags
   // and can not be NONE.
-  virtual void GetApproximateSizes(ColumnFamilyHandle* column_family,
-                                   const Range* ranges, int n, uint64_t* sizes,
-                                   uint8_t include_flags = INCLUDE_FILES) {
+  virtual Status GetApproximateSizes(ColumnFamilyHandle* column_family,
```

Comment:
Although this is not a breaking change, it is an API change (add to HISTORY). I was going to argue that these simplified versions of GetApproximateSizes are explicitly meant to ignore error and just "be inaccurate" in case of some problem. But on further inspection, I'd argue you're fixing an API bug, because `sizes` might not be set, at least in the InvalidArgument case.

Since this function is a pretty thin wrapper around the full Status-returning version, we could instead document that in some error cases `sizes` might not be over-written at all, and to use the full version if you want diagnostics. I think I slightly prefer this approach, especially if it's bundled with a giant PR of test updates.

Reply:
Okay I have now added a note to `HISTORY.md` about this.